web/cypress/e2e/app.cy.ts

```diff
@@ -0,0 +1,16 @@
+import "cypress-axe";
+
+describe("ARC Portal UI", () => {
+  it("shows the ARC portal phrase on the homepage", () => {
+    cy.visit("/");
+
+    cy.contains("ARC portal").should("be.visible");
+    cy.contains("fibble").should("not.exist");
+
+    // run accessibility check
+    cy.injectAxe();
+    cy.checkA11y(undefined, {
+      includedImpacts: ["critical", "serious"],
+    });
```

Comment:
There is a lot of customisation possible with the axe library. I made the default check focus on critical and serious accessibility flaws which seems like a reasonable place to start. But I'm open to opinions.

We might need to add more customisation soon as it seemed like the accessibility test failures could have been more descriptive in showing you what actually caused the failure. But let's see how we get on and we can add customisation if needed.

Reply:
sounds like a reasonable start. running this locally I get an error with `serious` 😞

Reply:
said error:
<img width="546" alt="Screenshot 2025-04-14 at 10 24 07" src="https://github.com/user-attachments/assets/4942cfd0-28d6-4d07-b7a8-39b5e46df2d9" />

Reply:
https://github.com/ucl-arc-tre/portal/pull/33/commits/00310c5459b71c6bf87e8b911a3792e1b91d9d47

This was an example of being caught out by brief flashes of content while the app is loading. The `login` button briefly flashes on load but disappears very quickly if you are already logged in. So the test was passing sometimes but occasionally the test was quick enough to catch the login button before it disappeared.

I updated the css to fix the accessibility issue. Try running it again.

Reply:
Ah, its still failing. I'll have another look.

Reply:
> briefly flashes on load but disappears very quickly if you are already logged in

flashiness rm-ed in https://github.com/ucl-arc-tre/portal/pull/32/files#diff-2e5aec0d80c33c218d97cb48edb61724a5e25bfc33478de5907916a90a92accf 😅

Reply:
I think it is now fixed here: https://github.com/ucl-arc-tre/portal/pull/33/commits/7c7adf307944a470b31bd154bcdb6d1a6e11556e

This is interesting. The issue is that the release version of the tests run in light mode whereas other environments run in dark mode (depending on your local settings probably). It just needed a bit of css adding to specifically target light mode to make the login button visible.

Reply:
That might warrant a bit of documentation to explain that, unless we want to make the tests more dynamic to run across both light and dark mode. @acholyn have you done that before?

Reply:
We could do something like this:

```
describe('Light mode test', () => {
  it('should render in light mode', () => {
    cy.setColorScheme('light');
    
    // run tests
  });
});
```

Did we decide we were going to support both light and dark mode in the end?

Reply:
I think light+dark mode accessibility tests would be the way to go, if indeed we're supporting both, which we probably are(?)